crates/red_knot/src/types/infer.rs

```diff
@@ -80,7 +80,18 @@ where
                     .resolve(ast.as_any_node_ref())
                     .expect("node key should resolve");
 
-                let ty = type_store.add_function(file_id, &node.name.id).into();
+                let decorator_tys: Vec<_> = node
+                    .decorator_list
+                    .iter()
+                    .map(|decorator| {
+                        infer_expr_type(db, file_id, &decorator.expression)
+                            .expect("decorator expression type should be inferrable")
```

Comment:
potention panic?

Reply:
This function (`infer_symbol_type`) also returns a `QueryResult<Type>`, so you can just use the `?` operator here to bubble up the error condition, if we get one, and unwrap the type.

I think currently the only error condition is `QueryCancelled`, meaning conditions (i.e. files) have changed and this query is already obsolete so we should stop working on it.

Reply:
Whoops, I forgot rust has maybe/either monads. Willfix :)